docs/case-studies/gitops.md

```diff
@@ -0,0 +1,143 @@
+---
+title:  GitOps with KubeVela
+---
+
+GitOps is a continuous delivery method that allows developers to automatically deploy applications by changing code in a Git repository. In this section, you will learn how to use KubeVela for GitOps deployment.
+
+## Preparation
+
+First, prepare a Git Repository with `Application` files that need to be applied, please refer to [KubeVela-GitOps-Demo](https://github.com/FogDong/KubeVela-GitOps-Demo):
+
+```yaml
+apiVersion: core.oam.dev/v1beta1
+kind: Application
+metadata:
+  name: first-vela-workflow
+  namespace: default
+spec:
+  components:
+  - name: express-server
+    type: webservice
+    properties:
+      image: stefanprodan/podinfo
+      port: 9898
+    traits:
+    - type: ingress
+      properties:
+        domain: testsvc.example.com
+        http:
+          /: 9898
+  - name: nginx-server
+    type: webservice
+    properties:
+      image: nginx:1.21
+      port: 80
+```
+
+Second, apply a secret with your Git username and password in cluster:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: my-secret
+type: kubernetes.io/basic-auth
+stringData:
+  username: your username
+  password: your password
+```
+
+## Write the Application that can apply automatically
+
+Apply the `Application` file to the cluster:
+
+```yaml
+apiVersion: core.oam.dev/v1beta1
+kind: Application
+metadata:
+  name: git-app
+spec:
+  components:
+  - name: gitops
+    type: kustomize
+    properties:
+      repoType: git
+      # your git repository url
+      url: https://github.com/FogDong/KubeVela-GitOps-Demo
+      # your secret name
+      secretRef: my-secret
+      # the interval time of pull config from repo
+      pullInterval: 1m
+      git:
+        branch: master
+      path: .
```

Comment:
We normally run `vela show xxx` to guide users how to fill the properties.

BTW, my definition is out-dated?

```
➜  kubevela.io git:(cloud-resource) ✗ vela show kustomize
I0908 14:57:10.474122   91708 request.go:668] Waited for 1.014010181s due to client-side throttling, not priority and fairness, request: GET:https://47.242.209.62:6443/apis/image.toolkit.fluxcd.io/v1alpha1?timeout=32s
# Properties
+--------------+------------------------------------------------------------------------------------------------------+--------+----------+---------+
|     NAME     |                                             DESCRIPTION                                              |  TYPE  | REQUIRED | DEFAULT |
+--------------+------------------------------------------------------------------------------------------------------+--------+----------+---------+
| path         | Path to the directory containing the kustomization.yaml file, or the set of plain YAMLs a            | string | true     |         |
|              | kustomization.yaml should be generated for.                                                          |        |          |         |
| pullInterval | The interval at which to check for repository updates.                                               | string | true     | 1m      |
| repoUrl      | The repository URL, can be a HTTP/S or SSH address.                                                  | string | true     |         |
| branch       | The Git reference to checkout and monitor for changes, defaults to master branch.                    | string | true     | master  |
+--------------+------------------------------------------------------------------------------------------------------+--------+----------+---------+
```

Reply:
I will update the kustomize in vela show later, thanks for the report